incubator/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaSubscribeFactory.java

```diff
@@ -698,93 +942,109 @@ private void onKafkaData(
             assert acknowledge <= sequence;
             assert sequence >= replySeq;
 
-            replySeq = sequence + reserved;
+            replySeq += reserved;
```

Comment:
Is this change correct?
Note: it seems to assume `sequence` doesn't advance, but the other side can move `sequence` forward if it wants to as well.

Reply:
I think this was necessary because of the buffered messages, as we don't increase the sequence at onKafkaData, but later.

Reply:
But if the reply sender advances to he sequence, this will break, right?